internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+	}
+}
+
+func (r *LibraryResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterId, libraryRep := parseId(req.ID)
+	waitParams := compute.Wait{
+		ClusterID: clusterId,
+		IsRefresh: true,
+	}
+	lib := LibraryExtended{}
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &lib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, lib)...)
+
+}
+
+func (r *LibraryResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	installLib := compute.InstallLibraries{
+		Libraries: []compute.Library{libGoSDK},
+	}
+	req.Plan.GetAttribute(ctx, path.Root("cluster_id"), &installLib.ClusterId)
+	err := w.Libraries.Install(ctx, installLib)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to install library", err.Error())
+		return
+	}
+	waitParams := compute.Wait{
+		ClusterID: installLib.ClusterId,
+		IsRunning: true,
+	}
+	libraryRep := libGoSDK.String()
+	installedLib := LibraryExtended{}
+
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &installedLib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, installedLib)...)
+}
+
+func (r *LibraryResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libraryTfSDK LibraryExtended
+	resp.Diagnostics.Append(req.State.Get(ctx, &libraryTfSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var libGoSDK compute.Library
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, libraryTfSDK, &libGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterId := libraryTfSDK.ClusterId.ValueString()
+	libraryRep := libGoSDK.String()
+	installedLib := LibraryExtended{}
+	waitParams := compute.Wait{
+		ClusterID: clusterId,
+		IsRefresh: true,
+	}
+
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &installedLib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, installedLib)...)
+}
+
+func (r *LibraryResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
I guess we need this implementation, but should this just fail?

Reply:
Yeah we should just make it fail